engine/sound/src/decoders/decoder_wav.cpp

```diff
@@ -109,12 +110,32 @@ namespace dmSoundCodec
         };
 
         struct DecodeStreamInfo {
+            DecodeStreamInfo() : m_IsADPCM(false) {}
+            virtual ~DecodeStreamInfo() = default;
+
             Info m_Info;
+            bool m_IsADPCM : 1;
+            uint8_t : 7;
             uint32_t m_Cursor;
             const void* m_Buffer;
             uint32_t m_BufferOffset;
             dmSound::HSoundData m_SoundData;
         };
+
+        struct DecodeStreamInfoADPCM : public DecodeStreamInfo {
+            DecodeStreamInfoADPCM() { m_IsADPCM = true; }
+            virtual ~DecodeStreamInfoADPCM() = default;
```

Comment:
Note that we're more of a C engine than C++. Also known as"c-like-c++". As such we prefer a `has-a` strategy as opposed to an `is-a`.

We only have class overrides in two places:
* where the 3rd party c++ libraries force us to
* type specializations for `dmHashTable` and `dmIndexPool`

I think this is the last change needed for this PR, and we're good to go.